miraclelinux/meta-debian-extended
# Add polkit package.

# Purpose of pull request

policykit

This pr adds policykit package based on meta-openembedded's /meta-oe/recipes-extended/polkit/polkit_0.104.bb

> When rebuilding the image after adding polkit to DISTRO_FEATURES, as described below, the dependency of DISTRO_FEATURES on polkit will cause the
> The systemd recipe that will be re-run from do_configure will cause an error in the do_install task, so you will need to execute the following command and process to clear SSTATE_CACHE.

# Test
## How to test

Check with qemuarm64.

Since polkit is required to be specified in the DISTRO_FEATURES variable in building polkit, add the following items to conf/local.conf to build core-image-minimal.

```
echo 'DISTRO_FEATURES_append =  " polkit " ' >> conf/local.conf
echo 'IMAGE_INSTALL_append = " polkit " ' >> conf/local.conf
```

Start qemuarm and type the command.

- pk-example-frobnicate
- pkaction
- pkexec
- pkttyagent
- polkitd

## Test result

### pk-example-frobnicate
```
root@qemuarm64:~# pk-example-frobnicate
In pk-example-frobnicate
uid:           0
euid:          0
args:         `pk-example-frobnicate'
cwd:           /home/root
environment:
  USER=root
  SHLVL=1
  HOME=/home/root
  OLDPWD=/usr/bin
  HUSHLOGIN=FALSE
  PS1=\u@\h:\w\$
  LOGNAME=root
  TERM=vt102
  PATH=/usr/local/bin:/usr/bin:/bin:/usr/local/sbin:/usr/sbin:/sbin
  HZ=100
  SHELL=/bin/sh
  PWD=/home/root
  EDITOR=vi
```

### pkaction
```
root@qemuarm64:~# pkaction
org.freedesktop.policykit.example.pkexec.run-frobnicate
org.freedesktop.policykit.exec
org.freedesktop.policykit.lockdown
```

### pkexec
```
root@qemuarm64:~# pkexec --version
pkaction version 0.105
```

### pkttyagent
```
root@qemuarm64:~#　pkttyagent --version
pkttyagent version 0.105
```

### polkitd
```
root@qemuarm64:~# /usr/libexec/polkitd
Entering main event loop
Connected to the system bus
Registering null backend at priority -10
Error getting login monitor: -2Using authority class PolkitBackendLocalAuthority
Acquired the name org.freedesktop.PolicyKit1
^CHandling SIGINT
Shutting down
Exiting with code 0
```
